app/src/main/java/com/github/h3lp3rs/h3lp/forum/implementation/SimpleDBForum.kt

```diff
@@ -1,267 +1,268 @@
-        const val POSTS_LIST = "posts"
-    }
-}
+package com.github.h3lp3rs.h3lp.forum.implementation
+
+import android.os.Build
+import androidx.annotation.RequiresApi
+import com.github.h3lp3rs.h3lp.database.Database
+import com.github.h3lp3rs.h3lp.forum.*
+import com.github.h3lp3rs.h3lp.forum.data.ForumPostData
+import java.time.ZonedDateTime
+import java.util.concurrent.CompletableFuture
+
+/**
+ * This abstract forum behaves according to the ForumProtocol.md limitations (subset of Forum
+ * interface) and uses our key-value database as an underlying data structure.
+ *
+ * @param rootForum An implementation of the underlying database acting as root of our forum.
+ */
+abstract class SimpleDBForum(private val rootForum: Database) : Forum {
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    override fun newPost(author: String, content: String): CompletableFuture<ForumPost> {
+        var key = ""
+        // Incrementing by 2 so that a post's replies key is always 1 more than a post's key (this
+        // is also an optimization to avoid us requiring 2 calls to incrementAndGet)
+        return rootForum.incrementAndGet(UNIQUE_POST_ID, 2).thenApply { postKey ->
+            key = postKey.toString()
+            val repliesKey = (postKey + 1).toString()
+
+            val forumPostData = ForumPostData(
+                author,
+                content,
+                getFormattedPostTime(ZonedDateTime.now()),
+                key,
+                repliesKey,
+                getCurrentCategory()
+            )
+            rootForum.setObject(
+                pathToKey(path + key),
+                ForumPostData::class.java,
+                forumPostData
+            )
+
+            // If the forum is a category, this means that the post is a "main post" (not a reply)
+            // we thus add it to the posts in the database (as explained in the forum protocol)
+            if (isCategory()) {
+                rootForum.addToObjectsListConcurrently(
+                    pathToKey(path + POSTS_LIST),
+                    String::class.java,
+                    key
+                )
+            } else { // For mocking purposes this needs to be added
+                rootForum.addToObjectsListConcurrently(
+                    pathToKey(path),
+                    ForumPostData::class.java,
+                    forumPostData
+                )
+            }
+            // We can't use getPost here since we aren't sure that the setObject succeeded yet
+            ForumPost(this, forumPostData, emptyList())
+        }
+    }
+
+    /**
+     * Gets the post date in the following format day of month month hour-minutes
+     * @param currentTime The current date
+     * @return Formatted current date
+     */
+    @RequiresApi(Build.VERSION_CODES.O)
+    private fun getFormattedPostTime(currentTime: ZonedDateTime): String {
+        return currentTime.dayOfMonth.toString() + "" + currentTime.month.toString() + " " +

```

Comment:
\+ "" seems bizarre :monocle_face:

Reply:
Not my code, cherry picked from Wiam's commit :D

Reply:
@Ahayderi don't worry, I noticed it. It has been fixed on my PR